tests/trainers/java/files_and_directories.java

```diff
@@ -0,0 +1,185 @@
+/**
+Test to check the Java File module.
+TODO
+*/
+import java.io.*;
+import java.io.IOException;
+import java.util.UUID;
+
+class Test {
+    int step = 1;
+    String textContent = "Content of the text file";
+    int errors = 0;
+
+    private String createTextFile(String filepath) {
+        try {
+            FileWriter file = new FileWriter(filepath);
+            file.write(this.textContent);
+            file.close();
+            return "OK";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }
+    }
+
+    private String readTextFile(String filepath) {
+        try {
+            FileReader file = new FileReader(filepath);
+            char [] a = new char[50];
+            file.read(a);
```

Comment:
`read` returns the number of read characters, you can use this to avoid having to trim the output:

```suggestion
            final int n = file.read(a);
            final String output = new String(a, 0, n);
```

Also I would increase the size of the char array to something like 1024 - the malus is almost negligible, but you avoid future issues that arise from the buffer size being really really small.